packages/techdocs-common/src/stages/publish/types.ts

```diff
@@ -53,6 +61,14 @@ export type TechDocsMetadata = {
  * It also provides APIs to communicate with the storage service.
  */
 export interface PublisherBase {
+  /**
+   * Check if the configuration is valid. This check tries to perform certain checks to see if the
+   * publisher is configured correctly and can be used to publish or read documentations.
+   * The different implementations might e.g. use the provided service credentials to access the
+   * target or check if a folder/bucket is available.
+   */
+  validateConfiguration(): Promise<ConfigurationValidationResponse>;
```

Comment:
Now that I'm seeing this in context, and given we may implement additional checks at some later stage, `validateConfiguration()` feels a little misleading.  At a glance, I would imagine "validate configuration" to be more of a quick "does the config match the schema" or "do I have the minimum viable config keys set" type of check rather than a "I'm going to make network calls against an external service" type of check.

Perhaps `PublisherBase::getReadiness(): Promise<ReadinessResponse>`?  Or something similar.

And then in the context of "readiness" I guess instead of `isValid` (which is kind of config-y) could be `isAvailable`?

Reply:
👍  I updated it